### PR TITLE
Create MP JWT 2.1 feature files based off of PR #22083

### PR DESCRIPTION
**Pull request by jhanders34**
- Create mpJwt 2.1 public and API features
- Have mpJwt 2.1 not pull in appSecurity feature, but instead update servlet, restfulWS and CDI auto feature to include mpJwt 2.1

For issue #21186

### CODE REVIEW

**Comment by jhanders34** [comment]
#libby #build

File: dev/com.ibm.ws.security.fat.common/src/com/ibm/ws/security/fat/common/actions/SecurityTestFeatureEE10RepeatAction.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    public boolean isEnabled() {
+        Log.info(thisClass, "isEnabled", "testRunMode: " + testRunMode);
+        Log.info(thisClass, "isEnabled", "complexId: " + complexId);
+        Log.info(thisClass, "isEnabled", "Overall test mode: " + TestModeFilter.FRAMEWORK_TEST_MODE);
+        // allow if mode matches or mode not set
+        if (testRunMode != null && (TestModeFilter.FRAMEWORK_TEST_MODE != testRunMode)) {
+            Log.info(thisClass, "isEnabled", "Skipping action '" + toString() + "' because the test mode " + testRunMode +
+                    " is not valid for current mode " + TestModeFilter.FRAMEWORK_TEST_MODE);
+            return false;
+        }
```
Comment (Azquelt) [inline dev/com.ibm.ws.security.fat.common/src/com/ibm/ws/security/fat/common/actions/SecurityTestFeatureEE10RepeatAction.java:64]: This is not the normal check for whether a test should run in the given mode.

Looking at the places `SecurityTestFeatureEE9RepeatAction.liteFATOnly()` is used though, it appears to be deliberate?

Comment (jhanders34) [reply dev/com.ibm.ws.security.fat.common/src/com/ibm/ws/security/fat/common/actions/SecurityTestFeatureEE10RepeatAction.java:64]: It is close to what we do in the other code.  There is already lite FAT only option in the main code as well and could be change to use that function instead.  I just copied the EE9 implementation without changes.

Comment (Azquelt) [reply dev/com.ibm.ws.security.fat.common/src/com/ibm/ws/security/fat/common/actions/SecurityTestFeatureEE10RepeatAction.java:64]: Just seems odd that we have that option at all anywhere, given that FULL mode is supposed to be a superset of LITE.

Comment (jhanders34) [reply dev/com.ibm.ws.security.fat.common/src/com/ibm/ws/security/fat/common/actions/SecurityTestFeatureEE10RepeatAction.java:64]: The main problem is the slow build machines where running in full mode will cause things to go over 3 hours.  So some teams have chosen to run a subset of tests in lite mode only.  Likely this whole class could be removed and replaced with the main classes with existing function, but would have to investigate it.  I think that may be outside of the scope of this PR.

**Review by Azquelt — APPROVED** [review]

**Comment by Azquelt** [comment]
Everything looks sensible to me. I don't know enough about how the security components fit together to say whether the autofeature changes are correct, so it would be good for someone else to look at that.

**Comment by jhanders34** [comment]
#libby #build

**Review by teddyjtorres — APPROVED** [review]

**Review by c00crane — APPROVED** [review]
Thanks for the updates Jared - I didn't had a feature to work with.

**Review by ayoho — APPROVED** [review]

File: dev/com.ibm.websphere.appserver.features/visibility/public/mpJwt-2.1/resources/l10n/io.openliberty.mpJwt-2.1.properties [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+# Contributors:
+#     IBM Corporation - initial API and implementation
+###############################################################################
+#
+#ISMESSAGEFILE FALSE
+#NLS_ENCODING=UNICODE
+#NLS_MESSAGEFORMAT_NONE
+#
+
+description=This feature enables web applications or microservices to use the Eclipse MicroProfile JSON Web Token 2.1 specification to authenticate users instead of, or in addition to, the configured user registry.
```
Comment (hiltoncv) [inline dev/com.ibm.websphere.appserver.features/visibility/public/mpJwt-2.1/resources/l10n/io.openliberty.mpJwt-2.1.properties:17]: As informed by @jhanders34 only the message version is updated. ID reviewed the original message. ID review is not required for this message.

**Comment by jhanders34** [comment]
#libby #build

**Pull request merged**